app/services/media_storage/stored_path.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module MediaStorage
+  module StoredPath
+    # Construct the media access URL
+    # and ensure we remove the old aws s3 domain prefix in the stored paths
+    # but leave intact any non old s3 URL, i.e. new azure ones
+    def self.media_url(url, stored_path)
+      uri = URI("https://#{stored_path}")
+      PublicSuffix.parse(uri.host)
+      # we have a valid domain prefix here so remove it
+      # to allow us to construct the URL correctly
+      File.join(url, uri.path)
```

Comment:
We will need to remove the environment from the path and add `public`, which will specify the  public container. Something like this-
```suggestion
      uri_path = uri.path.delete_prefix('/'+ENV['RAILS_ENV'])
      uri_path = '/public' + uri_path
      File.join(url, uri_path)
```

Reply:
We will need to keep these in for the single `panoptes-uploads.zooniverse.org` CDN all our data sits behind (in production & staging) to distinguish between staging / production target paths that it proxies to.